### PR TITLE
feat: don't parse self/ctx parameter annotations of prefix command callbacks

### PR DESCRIPTION
**Pull request by shiftinv**
## Summary

`get_signature_parameters` would previously evaluate annotations of all parameters, including `self`/`ctx`, which doesn't allow for code like this:
```py
...
if TYPE_CHECKING:
    from .my_bot import MyBot

class SomeCog(Cog):
    @commands.command()
    async def cmd(self, ctx: Context["MyBot"], stuff: str):
        ...
```

Admittedly, that pattern may be rare, but due to this issue a workaround was required in #828.
This PR avoids parsing the annotations of the `self` (if present) and `ctx` parameter entirely, as they're not relevant for the process.

## Checklist

- [x] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [x] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by onerandomusername** [comment]
@shiftinv would you please resolve conflicts?

File: disnake/utils.py [hunk 1]
```diff
@@ -1331,6 +1332,45 @@ def assert_never(arg: Never, /) -> None:
     pass
 
 
+def signature_has_self_param(function: Callable[..., Any]) -> bool:
```
Comment (onerandomusername) [inline disnake/utils.py:1335]: this should also be used where we assume the parameter name is "self" in command parsing elsewhere.

Comment (shiftinv) [reply disnake/utils.py:1335]: sure: https://github.com/DisnakeDev/disnake/pull/847/commits/d9184e6fef7a13c46105711c123dde66fe2d7dac (+ other commits)
I'm a bit wary of changing core slash parts of slash right now, but having tested this change a bunch, it should be fine.

**Review by EQUENOS — APPROVED** [review]
Looks good, tests were successful

**Pull request merged**